File: libloginshare/src/main/java/com/hacknife/loginsharepay/proxy/ProxyFragment.java
package com.hacknife.loginsharepay.proxy;


import android.content.Intent;
import android.support.v4.app.Fragment;

import com.hacknife.loginsharepay.impl.BaseLoginShare;

/**
 * author  : Hacknife
 * e-mail  : 4884280@qq.com
 * github  : http://github.com/hacknife
 * project : LoginShare
 */

//替身Fragment，必须要V4的才行
public class ProxyFragment extends Fragment {


    private BaseLoginShare baseLogin;

    public ProxyFragment() {
    }

    public ProxyFragment(BaseLoginShare baseLogin) {
        this.baseLogin = baseLogin;
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        //QQ登录的回调
        if (baseLogin.getTencent() != null) {
            baseLogin.getTencent().onActivityResultData(requestCode, resultCode, data, baseLogin.getIUIListener());
        }
        //微博登录的回调
        if (baseLogin.getSsoHandler() != null) {
            baseLogin.getSsoHandler().authorizeCallBack(requestCode, resultCode, data);
        }
        super.onActivityResult(requestCode, resultCode, data);

    }

}

File: authority/src/main/java/com/hacknife/authority/proxy/ProxyFragment.kt
package com.hacknife.authority.proxy

import android.content.Intent
import androidx.fragment.app.Fragment

/**
 * author : 段泽全(hacknife)
 * e-mail : hacknife@outlook.com
 * time   : 2019/8/5
 * desc   : MVVM
 * version: 1.0
 */
class ProxyFragment : Fragment() {
    private var activityResult: ((requestCode: Int, resultCode: Int, data: Intent?) -> Unit)? = null
    private var destroy: (() -> Unit)? = null
    fun setActivityResult(result: (requestCode: Int, resultCode: Int, data: Intent?) -> Unit) {
        activityResult = result
    }

    fun setDestroy(l: (() -> Unit)) {
        destroy = l
    }

    companion object {
        const val TAG = "ProxyFragment"
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        activityResult?.invoke(requestCode, resultCode, data)
    }

    override fun onDestroy() {
        super.onDestroy()
        destroy?.invoke()
    }
}